src/nodes/typeGuards/compound.ts

```diff
@@ -61,6 +61,23 @@ export function isJsxTagNamePropertyAccess(
 	);
 }
 
+/**
+ * ts.NamedDeclaration but the name property isn't optional.
+ */
+export interface NamedDeclaration extends ts.NamedDeclaration {
+	name: ts.NamedDeclaration["name"] extends infer T | undefined
+		? T extends undefined
+			? never
+			: T
+		: never;
```

Comment:
This could just be declared as `name: NonNullable<ts.NamedDeclaration["name"]>`.

This will be less code and easier to maintain/understand, but will make the intelliSense report `node.name` in terms of `NonNullable` instead of a union of types. 

i.e

`NonNullable<ts.DeclarationName | undefined>`
instead of
`ts.Identifier | ts.StringLiteral | ts.PrivateIdentifier | ts.NoSubstitutionTemplateLiteral | ts.NumericLiteral | ts.ComputedPropertyName | ts.ElementAccessExpression | ts.ObjectBindingPattern | ts.ArrayBindingPattern | ts.PropertyAccessEntityNameExpression`

Reply:
Yeah I think the `NonNullable` type is fine, even better. Those long unions are hard to read through.